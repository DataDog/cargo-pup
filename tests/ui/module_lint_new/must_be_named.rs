// This product includes software developed at Datadog (https://www.datadoghq.com/) Copyright 2024 Datadog, Inc.

//@compile-flags: --crate-name test_must_be_named
//@compile-flags: --crate-type lib

// This module follows the required naming pattern (good_*)
pub mod good_module {
    pub fn valid_function() {
        println!("This is a valid module name following the pattern");
    }
}

// These modules don't follow the naming pattern and should trigger errors
pub mod incorrect_name { //~ ERROR: Module must match pattern 'good_*', found 'incorrect_name'
    pub fn some_function() {
        println!("This module doesn't follow the naming pattern");
    }
}

pub mod another_bad_name { //~ ERROR: Module must match pattern 'good_*', found 'another_bad_name'
    pub fn another_function() {
        println!("This module also doesn't follow the naming pattern");
    }
} 
