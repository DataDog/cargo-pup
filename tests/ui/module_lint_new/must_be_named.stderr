error: Module must match pattern 'good_*', found 'incorrect_name'
  --> tests/ui/module_lint_new/must_be_named.rs:12:1
   |
LL | / pub mod incorrect_name {
LL | |     pub fn some_function() {
LL | |         println!("This module doesn't follow the naming pattern");
LL | |     }
LL | | }
   | |_^
   |
   = help: Rename this module to match the pattern 'good_*'
   = note: Applied by cargo-pup rule 'must_be_named_test'.
   = note: `#[deny(module_must_be_named)]` on by default

error: Module must match pattern 'good_*', found 'another_bad_name'
  --> tests/ui/module_lint_new/must_be_named.rs:18:1
   |
LL | / pub mod another_bad_name {
LL | |     pub fn another_function() {
LL | |         println!("This module also doesn't follow the naming pattern");
LL | |     }
LL | | } 
   | |_^
   |
   = help: Rename this module to match the pattern 'good_*'
   = note: Applied by cargo-pup rule 'must_be_named_test'.

error: aborting due to 2 previous errors

